src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -305,7 +313,7 @@ def main():
 
     if args.submit_job:
         try:
-            task_mgr.submit_job(restart_on_demand=args.force)
+            task_mgr.submit_job()
         except PreconditionError as e:
```

Comment:
so submit_job can't throw a precondition error as is, but maybe both the constructor and the submit call should check preconditions in case there's a change in between. Either way the I assume the precondition error needs to be caught when calling the constructor

Reply:
👍 updated to catch preconditionError at both places.